app/components/classification-multi-select.js

```diff
@@ -0,0 +1,104 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { task } from 'ember-concurrency';
+import { trackedTask } from 'ember-resources/util/ember-concurrency';
+import { CENTRAL_WORSHIP_SERVICE_BLACKLIST } from 'frontend-organization-portal/models/recognized-worship-type';
+import { CLASSIFICATION_CODE } from 'frontend-organization-portal/models/administrative-unit-classification-code';
+
+export default class ClassificationMultiSelectComponent extends Component {
+  @service store;
+  @service currentSession;
+  classifications = trackedTask(this, this.loadClassificationsTask, () => [
+    this.args.selectedRecognizedWorshipTypeId,
+  ]);
+
+  get selectedClassifications() {
+    if (typeof this.args.selected === 'string') {
+      return this.findClassificationById(this.args.selected);
+    }
+
+    return this.args.selected;
+  }
+
+  findClassificationById(id) {
+    if (this.classifications.isRunning) {
+      return null;
+    }
+
+    let classifications = this.classifications.value;
+    return classifications.find((status) => status.id === id);
+  }
+
+  @task *loadClassificationsTask() {
+    // Trick used to avoid infinite loop
+    // See https://github.com/NullVoxPopuli/ember-resources/issues/340 for more details
+    yield Promise.resolve();
+
+    let allowedIds;
+
+    let selectedRecognizedWorshipTypeId =
+      this.args.selectedRecognizedWorshipTypeId;
+
+    if (
+      selectedRecognizedWorshipTypeId &&
+      this.isIdInBlacklist(selectedRecognizedWorshipTypeId)
+    ) {
+      allowedIds = [CLASSIFICATION_CODE.WORSHIP_SERVICE];
+    } else if (this.args.restrictForNewBestuursenheden) {
+      allowedIds = [
+        CLASSIFICATION_CODE.WORSHIP_SERVICE,
+        CLASSIFICATION_CODE.CENTRAL_WORSHIP_SERVICE,
+        CLASSIFICATION_CODE.DISTRICT,
+        CLASSIFICATION_CODE.AGB,
+        CLASSIFICATION_CODE.APB,
+        CLASSIFICATION_CODE.PROJECTVERENIGING,
+        CLASSIFICATION_CODE.DIENSTVERLENENDE_VERENIGING,
+        CLASSIFICATION_CODE.OPDRACHTHOUDENDE_VERENIGING,
+        CLASSIFICATION_CODE.OPDRACHTHOUDENDE_VERENIGING_MET_PRIVATE_DEELNAME,
+        CLASSIFICATION_CODE.POLICE_ZONE,
+        CLASSIFICATION_CODE.ASSISTANCE_ZONE,
+      ];
```

Comment:
This part is not relevant here as it's only used on creation of a new admin unit, which is in the normal `ClassificationSelect` component